src/chains/ethereum/ethereum/tests/api/debug/debug_storageRangeAt.test.ts

```diff
@@ -0,0 +1,487 @@
+          "0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563": {
+            key:
+              "0x0000000000000000000000000000000000000000000000000000000000000000",
+            value:
+              "0x0000000000000000000000000000000000000000000000000000000000000005"
+          },
+          "0x405787fa12a823e0f2b7631cc41b3ba8828b3321ca811111fa75cd3aa3bb5ace": {
+            key:
+              "0x0000000000000000000000000000000000000000000000000000000000000002",
+            value:
+              "0x68656c6c6f207270647200000000000000000000000000000000000000000014"
+          },
+          "0xb10e2d527612073b26eecdfd717e6a320cf44b4afac2b0732d9fcbe2b7fa0cf6": {
+            key:
+              "0x0000000000000000000000000000000000000000000000000000000000000001",
+            value:
+              "0x0000000000000000000000000000000000000000000000000000000000000001"
+          }
+        };
+
+        assert.deepStrictEqual(result.storage, storage);
+        assert.strictEqual(result.nextKey, null);
+      });
+
+      it("should return correct storage given different transaction indexes", async () => {
+        /* 
+          Strategy for this test:
+            1. Call miner.stop() so we can send a few transactions
+            2. Call miner.start() so we can mine all of the txs in a single block
+            3. Call debug_storageRangeAt for each transaction using the transaction index
+            4. Assert that the result will be the value we think it is in storage
+        */
+
+        await provider.send("eth_subscribe", ["newHeads"]);
+        await provider.send("miner_stop");
+
+        const tx1 = await provider.send("eth_sendTransaction", [
+          {
+            from: accounts[0],
+            to: contractAddress,
+            gas: 3141592,
+            data: `0x${methods["setValue(uint256)"]}0000000000000000000000000000000000000000000000000000000000000001`
+          }
+        ]);
+        const tx2 = await provider.send("eth_sendTransaction", [
+          {
+            from: accounts[0],
+            to: contractAddress,
+            gas: 3141592,
+            data: `0x${methods["setValue(uint256)"]}0000000000000000000000000000000000000000000000000000000000000002`
+          }
+        ]);
+        const tx3 = await provider.send("eth_sendTransaction", [
+          {
+            from: accounts[0],
+            to: contractAddress,
+            gas: 3141592,
+            data: `0x${methods["setValue(uint256)"]}0000000000000000000000000000000000000000000000000000000000000003`
+          }
+        ]);
+
+        await provider.send("miner_start");
+        await provider.once("message");
+
+        const txReceipt1 = await provider.send("eth_getTransactionReceipt", [
+          tx1
+        ]);
+        const txReceipt2 = await provider.send("eth_getTransactionReceipt", [
+          tx2
+        ]);
+        const txReceipt3 = await provider.send("eth_getTransactionReceipt", [
+          tx3
+        ]);
```

Comment:
a slightly neater way:

```suggestion
        const [txReceipt1, txReceipt2, txReceipt3] = await Promise.all([
          provider.send("eth_getTransactionReceipt", [tx1]),
          provider.send("eth_getTransactionReceipt", [tx2]),
          provider.send("eth_getTransactionReceipt", [tx3])
        ]);
```